snapcraft/plugins/_python/_python_finder.py

```diff
@@ -0,0 +1,44 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import glob
+
+
+def get_python_command(python_version, stage_dir, install_dir):
+    python_command = os.path.join('usr', 'bin', python_version)
+
+    # staged as in the stage dir, not from stage-packages
+    staged_python = os.path.join(stage_dir, python_command)
+    unstaged_python = os.path.join(install_dir, python_command)
+
+    if os.path.exists(staged_python):
+        return staged_python
+    else:
+        return unstaged_python
```

Comment:
what about having a safeguard here in case unstaged python doesn't exist? like in the next func.

Reply:
Yeah the usage of this function is a little odd, and perhaps it doesn't make as much sense in isolation. Check [this](https://github.com/snapcore/snapcraft/blob/master/snapcraft/plugins/python.py#L162) out, which uses it before any stage packages are pulled (i.e. before the part's python has even been fetched). This function is has two purposes: to figure out a python command to use, yes, but also to simply determine which area WILL contain python: staging, or the part. The latter assumption is the plugin's leaking into this module. If we modify this function to simply be the former (find an existing python), the Python plugin can catch the exception and make whatever assumptions it wants to make, which is a better separation of concerns, I think.

Reply:
Done.